primitives/manta/src/assets.rs

```diff
@@ -320,6 +323,8 @@ where
     fn ensure_valid(asset_id: AssetId) -> Result<(), FungibleLedgerError>;
 
     /// Check whether `account` can increase its balance by `amount` in the given `asset_id`.
+    /// Non-native assets will use the `can_increase_total_supply` check, while native assets will not.
+    /// Should be called before `deposit_can_mint`.
```

Comment:
is there any reason for it not to be called before `deposit_can_mint`?
If not, pls wrap them to ensure they'll always be called in order and make the current fns private